hub/src/server/server.ts

```diff
@@ -213,6 +214,21 @@ export class HubServer {
       }
     }
 
+    const requestTag = requestHeaders['if-match']
+    if (requestTag) {
```

Comment:
The DriveModel interface should have a new boolean property which indicates that the generic ETag validation should not be performed, and instead the driver will handle the check itself. Like `supportsETagConditionalUploads` or some better name. 

A couple use cases:
1. Improved latency (one less network request) for cloud storage APIs that support atomic conditional uploads using ETags. Azure supports this, so the PBC hub will have essentially no penalties when this goes live. Google Cloud API docs indicate it _might_ be supported (I can't tell without further testing). S3 protocol does not support it :(
2. Potential driver implementations that can perform a more efficient check for whatever reason. _Contrived example_: the `performStat` might be slow in order to retrieve the file size value, but really fast to check the etag.

Reply:
working on this!